pkg/urbit/jets/e/aes_siv.c

```diff
@@ -0,0 +1,356 @@
+/* j/5/aes_ecb.c
+**
+*/
+#include "all.h"
+
+#include <openssl/aes.h>
+
+#include "aes_siv.h"
+
+
+/* functions
+ */
+static void u3r_bytes_reverse(c3_w    a_w,
+                              c3_w    b_w,
+                              c3_y*   c_y,  /* out */
+                              u3_atom d)    /* in */
+{
+  u3r_bytes(a_w, b_w, c_y, d);
+  c3_w i_w;
+  for (i_w = 0; i_w < ((b_w - a_w) / 2) ; i_w++) {
+    c3_y lo = c_y[i_w];
+    c3_y hi = c_y[b_w - i_w - 1];
+    c_y[i_w] = hi;
+    c_y[b_w - i_w - 1] = lo;
+  }
+
+  return;
+}
+
+static u3_noun _siv_en(c3_y* key_y,
+                       c3_w keysize,
+                       u3_atom ads,
+                       u3_atom txt)
+{
+  AES_SIV_CTX* ctx = AES_SIV_CTX_new();
+  if ( 0 == ctx ) {
+    return u3m_bail(c3__fail);
+  }
+
+  if ( 0 == AES_SIV_Init(ctx, key_y, keysize) ) {
+    AES_SIV_CTX_free(ctx);
+    return u3m_bail(c3__exit);
```

Comment:
It's not at all clear to me that `AES_SIV_Init()` failing can only correspond to deterministic failures represented in the hoon/nock. If that's clear to you, can you add a comment explaining the failure conditions? And if it's not, please switch this to `c3__fail`.